src/bitswap/wantlist.js

```diff
@@ -5,42 +5,58 @@
 const { getDescribe, getIt, expect } = require('../utils/mocha')
 const { waitForWantlistKey } = require('./utils')
 
+const TIMEOUT = 120000
+
 module.exports = (createCommon, options) => {
   const describe = getDescribe(options)
   const it = getIt(options)
   const common = createCommon()
 
-  describe('.bitswap.wantlist', function () {
-    this.timeout(60 * 1000)
+  describe('.bitswap.wantlist', () => {
     let ipfsA
     let ipfsB
     const key = 'QmUBdnXXPyoDFXj3Hj39dNJ5VkN3QFRskXxcGaYFBB8CNR'
 
-    before(async () => {
+    before(async function () {
+      // CI takes longer to instantiate the daemon, so we need to increase the
+      // timeout for the before step
+      this.timeout(60 * 1000)
+
       ipfsA = await common.setup()
-      ipfsB = await common.setup()
+      ipfsB = await common.setup({ factoryOptions: { type: 'go' } })
 
       // Add key to the wantlist for ipfsB
-      ipfsB.block.get(key, () => {})
+      ipfsB.block.get(key).catch(() => {})
 
       await ipfsA.swarm.connect(ipfsB.peerId.addresses[0])
     })
 
-    after(() => common.teardown())
+    after(function () {
+      this.timeout(TIMEOUT)
 
-    it('should get the wantlist', () => {
-      return waitForWantlistKey(ipfsB, key)
+      return common.teardown()
     })
 
-    it('should get the wantlist by peer ID for a diffreent node', () => {
-      return waitForWantlistKey(ipfsA, key, { peerId: ipfsB.peerId.id })
+    it('should get the wantlist', function () {
+      this.timeout(TIMEOUT)
+      return waitForWantlistKey(ipfsB, key, { timeout: TIMEOUT })
```

Comment:
Do we need to pass the timeout here?  If the request doesn't finish in time the test will fail anyway and it's not supported by core, only the rpc client.